src/mlwm/build_inference_artifact.py

```diff
@@ -0,0 +1,350 @@
+        )
+
+        if datastore_name is not None:
+            # update the config path in the neural-lam config file to just have
+            # the name of the file
+            nl_config["datastores"][datastore_name][
+                "config_path"
+            ] = fn_datastore_config
+
+    fn_config = Path(nl_config_path).name
+    fp_config_dst = artifact_output_path / fn_config
+
+    # save the neural-lam config file
+    with open(fp_config_dst, "w") as f:
+        yaml.dump(nl_config, f)
+
+    logger.debug(f"Copied {nl_config_path} -> {fp_config_dst}")
+
+
+def _check_and_copy_training_cli_args(artifact_output_path: str):
+    """
+    Check for the presence of a file called `training_cli_args.yaml` in the
+    current directory, check that it is a valid yaml file, and copy it to the
+    artifact output directory.
+
+    Parameters
+    ----------
+    artifact_output_path : str
+        Path to the artifact output directory.
+    """
+    if not Path(TRAINING_CLI_ARGS_FILE).exists():
+        raise FileNotFoundError(
+            f"File {TRAINING_CLI_ARGS_FILE} not found. Please create this file "
+            "with the command line arguments that were used to train the "
+            "model."
+        )
+
+    try:
+        with open(TRAINING_CLI_ARGS_FILE, "r") as f:
+            yaml.safe_load(f)
+    except yaml.YAMLError as e:
+        raise ValueError(
+            f"File {TRAINING_CLI_ARGS_FILE} is not a valid yaml file. Please "
+            "check the file."
+        ) from e
+    # copy the file to the artifact output directory
+    shutil.copyfile(
+        TRAINING_CLI_ARGS_FILE,
+        Path(artifact_output_path) / TRAINING_CLI_ARGS_FILE,
+    )
+    logger.info(
+        f"Copied {TRAINING_CLI_ARGS_FILE} to "
+        f"{Path(artifact_output_path) / TRAINING_CLI_ARGS_FILE}"
+    )
+
+
+def _create_artifact_meta(artifact_output_path, nl_config_path, args):
+    """
+    Create a metadata file for the artifact. This includes the name of the
+    artifact, the version of the code, the path to the checkpoint file,
+    the path to the neural-lam config file, and the command line arguments
+    that were used to create the artifact.
+
+    Parameters
+    ----------
+    artifact_output_path : str
+        Path to the artifact output directory.
+    nl_config_path : str
+        Path to the neural-lam config file.
+    args : Namespace
+        Command line arguments that were used to create the artifact.
+    """
+
+    fp_artifact_meta_yaml = artifact_output_path / "artifact.yaml"
+
+    meta = dict(
+        artifact_name=artifact_output_path.name,
+        mlwm_version=__version__,
+        checkpoint_path=args.checkpoint,
+        config_path=nl_config_path,
+        argv=sys.argv,
+        # hostname
+        hostname=os.uname()[1],
+        # current working directory
+        cwd=os.getcwd(),
+        # python version
+        python_version=sys.version,
+        # mllam-data-pipeline version
```

Comment:
This is really the `mllam-data-prep` version and not pipeline?

Reply:
which "pipeline" are you referring to ? :) It is the `mllam-data-prep` version since we import it as `mdp` and use `mdp.__version__`

Reply:
Just the comment: `# mllam-data-pipeline version`

Reply:
Fixed in https://github.com/leifdenby/mlwm-deployment/commit/f0e739f723e7d77e531b2800e56b7f5d485b3518 🤣